server/src/infra/ldap_handler.rs

```diff
@@ -2772,6 +3012,97 @@ mod tests {
+            base: "cn=Subschema".to_string(),
+            scope: LdapSearchScope::Base,
+            aliases: LdapDerefAliases::Never,
+            sizelimit: 0,
+            timelimit: 0,
+            typesonly: false,
+            filter: LdapFilter::Present("objectClass".to_string()),
+            attrs: vec!["supportedExtension".to_string()],
+        };
+
+        let current_time_utc = Utc::now().format("%Y%m%d%H%M%SZ").to_string().into_bytes();
+
+        let expected_response: Vec<LdapOp> = vec![
+            LdapOp::SearchResultEntry(LdapSearchResultEntry {
+                dn: "cn=Subschema".to_owned(),
+                attributes: vec![LdapPartialAttribute {
+                    atype: "structuralObjectClass".to_owned(),
+                    vals: vec![b"subentry".to_vec()]
+                },
+                LdapPartialAttribute {
+                    atype: "objectClass".to_owned(),
+                    vals: vec![b"top".to_vec(), b"subentry".to_vec(), b"subschema".to_vec(), b"extensibleObject".to_vec()]
+                },
+                LdapPartialAttribute {
+                    atype: "cn".to_owned(),
+                    vals: vec![b"Subschema".to_vec()]
+                },
+                LdapPartialAttribute {
+                    atype: "createTimestamp".to_owned(),
+                    vals: vec![current_time_utc.to_vec()]
+                },
+                LdapPartialAttribute {
+                    atype: "modifyTimestamp".to_owned(),
+                    vals: vec![current_time_utc.to_vec()]
+                },
+                LdapPartialAttribute {
+                    atype: "ldapSyntaxes".to_owned(),
+                    vals: vec![
+                        b"( 1.3.6.1.4.1.1466.115.121.1.15 DESC 'Directory String' )".to_vec(),
+                        b"( 1.3.6.1.4.1.1466.115.121.1.24 DESC 'Generalized Time' )".to_vec(),
+                        b"( 1.3.6.1.4.1.1466.115.121.1.27 DESC 'Integer' )".to_vec(),
+                        b"( 1.3.6.1.4.1.1466.115.121.1.28 DESC 'JPEG' X-NOT-HUMAN-READABLE 'TRUE' )".to_vec()
+                    ]
+                },
+                LdapPartialAttribute {
+                    atype: "attributeTypes".to_owned(),
+                    vals: vec![
+                        b"( 2.0 NAME 'String' SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 )".to_vec(),
+                        b"( 2.1 NAME 'Integer' SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 )".to_vec(),
+                        b"( 2.2 NAME 'JpegPhoto' SYNTAX 1.3.6.1.4.1.1466.115.121.1.28 )".to_vec(),
+                        b"( 2.3 NAME 'DateTime' SYNTAX 1.3.6.1.4.1.1466.115.121.1.24 )".to_vec(),
+                        b"( 2.4 NAME 'avatar' DESC 'LLDAP: builtin attribute' SUP JpegPhoto )".to_vec(),
+                        b"( 2.5 NAME 'creation_date' DESC 'LLDAP: builtin attribute' SUP DateTime )".to_vec(),
+                        b"( 2.6 NAME 'display_name' DESC 'LLDAP: builtin attribute' SUP String )".to_vec(),
```

Comment:
How does it work if users and groups have the same attribute name, but with different types? Should that be forbidden?